gluten-core/src/main/java/io/glutenproject/vectorized/JniWorkspace.java

```diff
@@ -29,6 +30,13 @@ private JniWorkspace(String rootDir) {
       this.jniLibLoader = new JniLibLoader(workDir);
       this.jniResourceHelper = new JniResourceHelper(workDir);
       LOG.info("JNI workspace {} created in root directory {}", workDir, rootDir);
+      Runtime.getRuntime().addShutdownHook(new Thread(() -> {
```

Comment:
Can you add it to `GlutenShutdownManager`

Reply:
Nice solution.